deploy.py

```diff
@@ -535,19 +602,36 @@ def main():
     while args.region == 'list':
       region_list = get_gcp_regions(credentials, project_id)
       print('\n'.join(region_list))
-      args.region = (
-          input(
-              'Which region is the GA export in (list for a list of regions)? '
-          ).strip())
+      args.region = (input(
+          'Which region is the GA export in (list for a list of regions)? ').
+                     strip())
   if not args.ga_property:
-    args.ga_property = (
-        input(
-            'Please enter the GA property ID you are collecting CWV data with: '
-        ).strip())
+    args.ga_property = (input(
+        'Please enter the GA property ID you are collecting CWV data with: ').
+                        strip())
     if not args.ga_property.isdigit():
       raise SystemExit('Only GA4 properties are supported at this time.')
 
-  deploy_scheduled_materialize_query(project_id, args.region, args.ga_property)
+  if not args.iam_service_account:
+    if (hasattr(credentials, 'service_account_email') and
+        credentials.service_account_email == 'default'):
+      args.iam_service_account = get_default_service_account_email(
+          project_id, credentials)
+    else:
+      args.iam_service_account = credentials.service_account_email
```

Comment:
Entering `else` block causes an exception in my case because my `credentials` object doesn't have attribute `service_account_email`

Reply:
@adamread I think the logic should be as following, let me know if I misunderstood - https://miro.com/app/board/uXjVPLw_y78=/

Reply:
I think I've fixed it now. I've one with just creating and adding the role. Trying to find out if the service account has the permissions is not as easy as you'd think it should be.